src/main/java/com/dre/brewery/integration/listeners/movecraft/TranslationListener.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * BreweryX Bukkit-Plugin for an alternate brewing process
+ * Copyright (C) 2024-2025 The Brewery Team
+ *
+ * This file is part of BreweryX.
+ *
+ * BreweryX is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BreweryX is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BreweryX. If not, see <http://www.gnu.org/licenses/gpl-3.0.html>.
+ */
+
+package com.dre.brewery.integration.listeners.movecraft;
+
+import com.dre.brewery.Barrel;
+import com.dre.brewery.utility.BoundingBox;
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.events.CraftTranslateEvent;
+import net.countercraft.movecraft.util.MathUtils;
+import net.countercraft.movecraft.util.hitboxes.HitBox;
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+
+public class TranslationListener implements Listener {
+    @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+    public void translateListener(CraftTranslateEvent e) {
+        Vector v = delta(e);
+        if (v == null)
+            return;
+
+        HitBox hitBox = e.getCraft().getHitBox();
+        for (Barrel barrel : Barrel.barrels) {
+            Location location = barrel.getSpigot().getLocation().clone();
+            MovecraftLocation mvLocation = MathUtils.bukkit2MovecraftLoc(location);
```

Comment:
```suggestion
            MovecraftLocation spigotLocation= MathUtils.bukkit2MovecraftLoc(barrel.getSpigot().getLocation());
```
Why clone it again, it's already done when retrieving it from a block. Also descriptive names